### PR TITLE
timeout and fail on no message from kafka consumer

### PR DESCRIPTION
**Pull request by aperrot42**
- fail if the consumer is not able to receive messages from kafka for more than `kafka.Config().Consumer.MaxProcessingTime`

 - fail if there is a kafka connection error

 - report errors from kafka consumer

### CODE REVIEW

File: kafka-sync.go [hunk 1]
```diff
@@ -207,6 +208,7 @@ func (s *Syncer) ApplyChanges(changes <-chan diff.Change, send func(KeyValue), s
 }
 
 func (s *Syncer) IndexTopic(kafka sarama.Client, index diff.Index) (msgCount uint64, err error) {
+
```
Comment (mcluseau) [inline kafka-sync.go]: Remove

File: kafka-sync.go [hunk 2]
```diff
@@ -257,12 +259,12 @@ func (s *Syncer) IndexTopic(kafka sarama.Client, index diff.Index) (msgCount uin
 
 	consumer, err := sarama.NewConsumerFromClient(kafka)
 	if err != nil {
-		return
+		return 0, err
```
Comment (mcluseau) [inline kafka-sync.go]: Return is enough

File: kafka-sync.go [hunk 3]
```diff
@@ -257,12 +259,12 @@ func (s *Syncer) IndexTopic(kafka sarama.Client, index diff.Index) (msgCount uin
...
 	}
 
 	pc, err := consumer.ConsumePartition(topic, partition, resumeOffset)
 	if err != nil {
-		return
+		return 0, err
```
Comment (mcluseau) [inline kafka-sync.go]: Return is enough

File: kafka-sync.go [hunk 4]
```diff
@@ -285,6 +287,7 @@ func (s *Syncer) IndexTopic(kafka sarama.Client, index diff.Index) (msgCount uin
 	go doIndex()
 
 	saveBatch := func(restart bool) {
+		glog.V(4).Info("finalize")
```
Comment (mcluseau) [inline kafka-sync.go]: Below comment plz ;)

File: kafka-sync.go [hunk 5]
```diff
@@ -299,31 +302,54 @@ func (s *Syncer) IndexTopic(kafka sarama.Client, index diff.Index) (msgCount uin
...
-			highWater = hw
-		}
-		glog.V(4).Info("-> offset: ", m.Offset, " / ", highWater-1)
-
-		value := m.Value
-		if bytes.Equal(value, s.RemovedValue) {
-			value = nil
+	// report kafka errors
+	go func() {
+		for m := range pc.Errors() {
```
Comment (mcluseau) [inline kafka-sync.go:308]: Add comment // FIXME fail on first error

**Review by mcluseau — CHANGES_REQUESTED** [review]

File: kafka-sync.go (hunk 3 above)
Comment (aperrot42) [reply kafka-sync.go]: indeed, changing that.

File: kafka-sync.go (hunk 5 above)
Comment (aperrot42) [reply kafka-sync.go:308]: did that and will try to work on this specific FIXME after working on an amazing sharepoint2sync middleware app.

File: kafka-sync.go (hunk 4 above)
Comment (aperrot42) [reply kafka-sync.go]: do you mean use finalize logic on kafka error ?

Comment (mcluseau) [reply kafka-sync.go]: No just move the log below the comment ;-)

**Comment by aperrot42** [comment]
Alright @mcluseau I think I've resolved the items ou pointed out. I'll integreate this version to sync2kafka as soon as the pr is accepted.

**Comment by mcluseau** [comment]
ok just squash all your commits and we're good to go ;)

**Comment by aperrot42** [comment]
alright @mcluseau squashed pr into 1 commit. Thanks for reviewing ;-)

**Pull request merged**